--- openssl-1.0.1j-clean/apps/apps.c	2014-10-15 14:53:39.000000000 +0200
+++ openssl-1.0.1j-amiga/apps/apps.c	2014-12-03 17:19:46.975009469 +0100
@@ -2888,7 +2888,7 @@
 	return (ret);
 	}
 
-#elif defined(OPENSSL_SYS_NETWARE)
+#elif defined(OPENSSL_SYS_NETWARE) || defined(OPENSSL_SYS_AMIGA)
 #include <time.h>
 
 double app_tminterval(int stop,int usertime)
@@ -2998,7 +2998,6 @@
 
 	return (ret);
 	}
-
 #else
 #include <sys/time.h>
 #include <sys/resource.h>
