--- openssl-1.0.1j-clean/crypto/bio/bss_fd.c	2014-10-15 14:51:06.000000000 +0200
+++ openssl-1.0.1j-amiga/crypto/bio/bss_fd.c	2014-12-03 17:19:54.438915580 +0100
@@ -61,7 +61,7 @@
 #define USE_SOCKETS
 #include "cryptlib.h"
 
-#if defined(OPENSSL_NO_POSIX_IO)
+#if defined(OPENSSL_NO_POSIX_IO) && !defined (OPENSSL_SYS_AMIGA)
 /*
  * One can argue that one should implement dummy placeholder for
  * BIO_s_fd here...
