--- openssl-1.0.1j-clean/e_os.h	2014-10-15 14:53:39.000000000 +0200
+++ openssl-1.0.1j-amiga/e_os.h	2014-12-03 17:49:00.236969869 +0100
@@ -108,6 +108,16 @@
 # endif
 #endif
 
+#if defined(OPENSSL_SYS_AMIGA)
+# define NO_FORK
+# define NO_CHMOD
+# define NO_SYSLOG
+# undef  DEVRANDOM
+# define DEVRANDOM "RANDOM:"
+# undef  DEVRANDOM_EGD
+# undef  OPENSSL_UNISTD
+#endif
+
 /********************************************************************
  The Microsoft section
  ********************************************************************/
@@ -451,6 +461,16 @@
 #    define LIST_SEPARATOR_CHAR ';'
 #    define EXIT(n)  { if (n) printf("ERROR: %d\n", (int)n); exit(n); }
 
+#  elif defined(OPENSSL_SYS_AMIGA)
+
+#    define OPENSSL_CONF	"openssl.cnf"
+#    define SSLEAY_CONF		OPENSSL_CONF
+#    define RFILE		"def.rnd"
+#    define DEFAULT_HOME	"RAM:T"
+#    define LIST_SEPARATOR_CHAR '|'
+#    define NUL_DEV		"NIL:"
+#    define EXIT(n)		exit(n)
+
 #  else
      /* !defined VMS */
 #    ifdef OPENSSL_SYS_MPE
@@ -490,7 +510,6 @@
 
 #endif
 
-
 /*************/
 
 #ifdef USE_SOCKETS
