--- openssl-1.0.1j-clean/crypto/opensslconf.h	2014-12-04 15:13:14.304876279 +0100
+++ openssl-1.0.1j-amiga/crypto/opensslconf.h	2014-12-04 15:37:39.302458975 +0100
@@ -8,6 +8,9 @@
 #ifndef OPENSSL_SYSNAME_SUNOS
 # define OPENSSL_SYSNAME_SUNOS
 #endif
+#ifndef OPENSSL_SYS_AMIGA
+# define OPENSSL_SYS_AMIGA
+#endif
 #ifndef OPENSSL_DOING_MAKEDEPEND
 
 
@@ -50,6 +53,9 @@
 #ifndef OPENSSL_NO_DYNAMIC_ENGINE
 # define OPENSSL_NO_DYNAMIC_ENGINE
 #endif
+#ifndef OPENSSL_NO_POSIX_IO
+# define OPENSSL_NO_POSIX_IO
+#endif
 
 /* The OPENSSL_NO_* macros are also defined as NO_* if the application
    asks for it.  This is a transient feature that is provided for those
@@ -90,7 +96,10 @@
 /* Generate 80386 code? */
 #undef I386_ONLY
 
-#if !(defined(VMS) || defined(__VMS)) /* VMS uses logical names instead */
+#if defined(OPENSSL_SYS_AMIGA)
+#define ENGINESDIR "ENVARC:ssl"
+#define OPENSSLDIR "ENVARC:ssl"
+#elif !(defined(VMS) || defined(__VMS)) /* VMS uses logical names instead */
 #if defined(HEADER_CRYPTLIB_H) && !defined(OPENSSLDIR)
 #define ENGINESDIR "/usr/local/ssl/lib/engines"
 #define OPENSSLDIR "/usr/local/ssl"
