--- openssl-1.0.1j-clean/apps/speed.c	2014-10-15 14:53:39.000000000 +0200
+++ openssl-1.0.1j-amiga/apps/speed.c	2014-12-03 17:10:44.169833452 +0100
@@ -210,7 +210,7 @@
 #endif
 
 #ifndef HAVE_FORK
-# if defined(OPENSSL_SYS_VMS) || defined(OPENSSL_SYS_WINDOWS) || defined(OPENSSL_SYS_MACINTOSH_CLASSIC) || defined(OPENSSL_SYS_OS2) || defined(OPENSSL_SYS_NETWARE)
+# if defined(OPENSSL_SYS_VMS) || defined(OPENSSL_SYS_WINDOWS) || defined(OPENSSL_SYS_MACINTOSH_CLASSIC) || defined(OPENSSL_SYS_OS2) || defined(OPENSSL_SYS_NETWARE) || defined(OPENSSL_SYS_AMIGA)
 #  define HAVE_FORK 0
 # else
 #  define HAVE_FORK 1
@@ -275,6 +275,10 @@
 static int rnd_fake = 0;
 #endif
 
+#if defined(OPENSSL_SYS_AMIGA)
+#undef SIGALRM
+#endif
+
 #ifdef SIGALRM
 #if defined(__STDC__) || defined(sgi) || defined(_AIX)
 #define SIGRETTYPE void
