--- openssl-1.0.1j-clean/crypto/uid.c	2014-07-22 21:41:23.000000000 +0200
+++ openssl-1.0.1j-amiga/crypto/uid.c	2014-12-03 17:01:09.765055501 +0100
@@ -65,7 +65,7 @@
 	return issetugid();
 	}
 
-#elif defined(OPENSSL_SYS_WIN32) || defined(OPENSSL_SYS_VXWORKS) || defined(OPENSSL_SYS_NETWARE)
+#elif defined(OPENSSL_SYS_WIN32) || defined(OPENSSL_SYS_VXWORKS) || defined(OPENSSL_SYS_NETWARE) || defined(OPENSSL_SYS_AMIGA)
 
 int OPENSSL_issetugid(void)
 	{
